bridge/client/app/_views/ktb-settings-view/ktb-integration-view/ktb-integration-view.component.ts

```diff
@@ -1,158 +1,102 @@
+import { distinctUntilChanged, filter, finalize, map, tap } from 'rxjs/operators';
 import { UniformRegistrationLog } from '../../../../../shared/interfaces/uniform-registration-log';
 import { UniformRegistration } from '../../../_models/uniform-registration';
 import { Location } from '@angular/common';
-import { UniformSubscription } from '../../../_models/uniform-subscription';
+
+export type Params = { projectName: string; integrationId?: string };
+
+const sortConfig: Record<string, (u: UniformRegistration) => string> = {
+  host: (u) => u.metadata.hostname,
+  namespace: (u) => u.metadata.kubernetesmetadata.namespace,
+  location: (u) => u.metadata.location,
+};
+
+const empty = '--EMPTY--';
 
 @Component({
   selector: 'ktb-keptn-services-list',
   templateUrl: './ktb-integration-view.component.html',
   styleUrls: ['./ktb-integration-view.component.scss'],
 })
-export class KtbIntegrationViewComponent implements OnInit, OnDestroy {
-  private readonly unsubscribe$ = new Subject<void>();
-  private selectedUniformRegistrationId$ = new Subject<string>();
-  private uniformRegistrationLogsSubject = new BehaviorSubject<UniformRegistrationLog[]>([]);
-  public UniformRegistrationClass = UniformRegistration;
+export class KtbIntegrationViewComponent {
+  private selectUniformRegistrationId$ = new Subject<string>();
   public isLoadingUniformRegistrations = true;
-  public uniformRegistrations: DtTableDataSource<UniformRegistration> = new DtTableDataSource();
-  public selectedUniformRegistration?: UniformRegistration;
-  public uniformRegistrationLogs$: Observable<UniformRegistrationLog[]> =
-    this.uniformRegistrationLogsSubject.asObservable();
   public isLoadingLogs = false;
-  public projectName?: string;
   public lastSeen?: Date;
+  private uniformRegistrations: DtTableDataSource<UniformRegistration> = new DtTableDataSource();
+
+  public params$: Observable<Params> = this.route.paramMap.pipe(
+    filter((paramMap) => !!paramMap.get('projectName')),
+    map((paramMap) => ({
+      projectName: paramMap.get('projectName') ?? empty,
+      integrationId: paramMap.get('integrationId') ?? undefined,
+    }))
+  );
+
+  private registrations$ = this.dataService
+    .getUniformRegistrations()
+    .pipe(finalize(() => (this.isLoadingUniformRegistrations = false)));
+
+  public uniformRegistrations$ = this.registrations$.pipe(
+    map((registrations) => {
+      this.uniformRegistrations.data = registrations;
+      return this.uniformRegistrations;
+    })
+  );
+
+  public selectedUniformRegistrationId$ = merge(
+    this.params$.pipe(map((params) => params.integrationId ?? empty)),
+    this.selectUniformRegistrationId$.asObservable()
+  ).pipe(distinctUntilChanged(), shareReplay(1));
+
+  public selectedUniformRegistration$ = this.selectedUniformRegistrationId$.pipe(
+    combineLatestWith(this.registrations$),
+    map(([regId, registrations]) => (regId ? registrations.find((r) => r.id === regId) : undefined))
```

Comment:
With `const empty` we would check all registrations for a not existing id.

Reply:
Fixed